models/_sector/dex/trades/dex_trades_beta.sql

```diff
@@ -0,0 +1,31 @@
+{{ config(
+    tags=['dunesql'],
+    schema = 'dex',
+    alias ='trades_beta',
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'project', 'version', 'tx_hash', 'evt_index']
+    )
+}}
+
+-- (blockchain, project, project_version, model)
+{% set base_models = [
+     ('ethereum',   'defiswap',   '1',    ref('defiswap_ethereum_base_trades'))
+    ,('ethereum',   'uniswap',    '1',    ref('uniswap_ethereum_v1_base_trades'))
+    ,('ethereum',   'uniswap',    '2',    ref('uniswap_ethereum_v2_base_trades'))
+    ,('ethereum',   'uniswap',    '3',    ref('uniswap_ethereum_v3_base_trades'))
+] %}
+
+
+-- macros/models/sector/dex
+{{
+    dex_enrich_trades(
+        blockchain = 'ethereum'
+        ,models = base_models
+        ,transactions_model = source('ethereum', 'transactions')
```

Comment:
same here, need to make it:
`,transactions_model = source('base_models[0]', 'transactions')`
but it failed `dbt compile`, since i'm already in a variable i assume? how to nest them?

Reply:
I think the dbt compile failed because as base_models is a list of dex.trades, which is more like a 2D array. It usually needs to work with jinja loop.